### ISSUE TITLE
21.05 Feature Freeze

### ISSUE THREAD

**Issue by jonringer** [NixOS/nixpkgs#121972]
Pinging all language, framework, and ecosystem owners to consolidate feature freeze items for the 21.05 release.

Please mention any items you see blocking the 21.05 release in your given domains. The branch off date will be the on the 21st of May. So there is still some time to address these items.

Nix/nix-cli ecosystem: @edolstra @grahamc @nbp @Profpatsch
Mobile: @samueldr
Nixos Modules / internals : @Infinisil @matthewbauer @Ericson2314 @orivej
Nixos tests: @tfc
Marketing: @garbas 

Emacs: @adisbladis
Erlang: @gleber @NixOS/beam 
Go: @kalbasit @Mic92 @zowoq
Haskell: @NixOS/haskell  
Python: @FRidh
Perl: @stigtsp
php: @NixOS/php
Ruby: @alyssais
rust: @bhipple @Mic92 @andir @LnL7

Darwin: @NixOS/darwin-maintainers

bazel: @mboes
blockchains @mmahut
podman: @NixOS/podman
Gnome: @jtojnar @NixOS/gnome
Qt / KDE: @ttuegel @NixOS/qt-kde 
Postgres: @thoughtpolice

in case I forgot anyone: @NixOS/nixpkgs-committers 

Anyone is free to propose potential blockers, but I would ask that you remember that this is a volunteer organization. Unless someone is likely to "pick up" the work and address the concern in the coming weeks, please only state critical issues. Or if anyone is active in a given ecosystem and I did not mention them, then they are free to state that there's unlikely to be any concerns as well.

**Comment by samueldr** [comment]
Not blocking per se, but would need to be merged in case it's forgotten.

 - #121450
 - #121720
 - #121834

The one about the Raspberry Pi image would need *something else on stable* if for any reason we cannot land it before the branch-off. Well, it'd need to be applied as it is on 21.05 even if we ended up not wanting it right now in unstable.

The other two I intend to merge by next week if it's not done beforehand. Neither of which are breaking changes or new features, mainly correctness changes for ARM installers.

**Comment by tomberek** [comment]
My goal is to get the Sourcehut service in: 
- 113244

**Comment by nh2** [comment]
I would like to get WireGuard DNS refresh merged: https://github.com/NixOS/nixpkgs/pull/121331

**Comment by samuelgrf** [comment]
I'd like to get QuickBMS (https://github.com/NixOS/nixpkgs/pull/121523) merged, I want to use it to package software that's only available in Windows installers.

**Comment by rnhmjoj** [comment]
It would be a good thing to fix #121293 before the freeze.

**Comment by danieldk** [comment]
It would also be good to fix https://github.com/NixOS/nixpkgs/issues/121994 before the freeze and/or definitely before the release. We have potentially broken Cargo hashes in Rust-based derivations.

**Comment by jtojnar** [comment]
GNOME 40 is likely ready, just waiting for a confirmation from @LEXUGE whether the patch from Fedora unbroke log-in with a fingerprint reader (and the other critical issue).

**Comment by augu5te** [comment]
Not a blocking one  #118573 . It's all about to have a clean OverlayFs multiple lower layers support.

**Comment by happysalada** [comment]
for erlang and elixir related stuff, feel free to mention the beam team.
I don't see anything blocking. There are a couple of interesting things being worked on #121555 but it won't be ready before the release I think.

**Comment by LEXUGE** [comment]
> GNOME 40 is likely ready, just waiting for a confirmation from @LEXUGE whether the patch from Fedora unbroke log-in with a fingerprint reader (and the other critical issue).

As I mentioned in the PR, the patch fixed the problem of not being able to key in passwords. However, it still doesn't log me in with fingerprint. Not sure if it qualifies as a fix.

**Comment by asymmetric** [comment]
I would like to get [this](https://github.com/NixOS/nixpkgs/pull/119540) in, which fixes:

```
nixos-rebuild switch --target-host monitor --build-host monitor --flake .#monitor
don't know how to build these paths:
  /nix/store/1km1lnysmjmbmad94l6bhm8v6x9xr9bm-nixos-system-monitor-20.09.20210123.a058d00
error: path '/nix/store/1km1lnysmjmbmad94l6bhm8v6x9xr9bm-nixos-system-monitor-20.09.20210123.a058d00' does not exist and cannot be created
```

**Comment by jonringer** [comment]
> for erlang and elixir related stuff, feel free to mention the beam team.
> I don't see anything blocking. There are a couple of interesting things being worked on #121555 but it won't be ready before the release I think.

New packages can always be backported, but this should be easy enough to get in :)

**Comment by jonringer** [comment]
> If systemd were to be updated to v248, another issue I'd try to fix before the freeze is: #73095. This is to avoid an ominous warning message on each boot.

After 20.09's issues with a systemd major bump, I'm convinced that the Desktop managers will have many issues moving to a version they didn't develop against. Like @vcunat said, the window for release-critical breaking changes was a few weeks ago. If we did have regressions, we would only have one staging cycle to fix them all before branch off.

**Comment by DianaOlympos** [comment]
Non blocking, but i want to drop some of the oldest erlang we support. They are not supported anymore. Will do a PR later this weekend.

**Comment by Technical27** [comment]
#101597 seems to be forgotten.
It is required for mangohud version >0.4 to be in nixpkgs. mangohud is currently on 0.6.1, while only 0.4.1 is in nixpkgs. vkBasalt isn't in nixpkgs due to the same reason.

**Comment by risicle** [comment]
I'd love to get #110893 reviewed so we can have gdb debugging for cpython in for 21.05.

**Comment by Mic92** [comment]
Regarding rust we will likely merge rust 1.52 before that into staging (https://github.com/NixOS/nixpkgs/pull/122110). The [changelog](https://blog.rust-lang.org/2021/05/06/Rust-1.52.0.html) was quite short so I don't think we will see regressions

**Comment by AndersonTorres** [comment]
I will spend some time upgrading the packages Repology is complaining.

**Comment by kinnison** [comment]
I'd like to propose #121072 to go into 21.05 because otherwise we'd be releasing with known out-of-date vscode python support.

**Comment by lukegb** [comment]
#122141 is an ffmpeg_4 bump (4.3->4.4), which might be worthwhile landing because it includes a bunch of AV1/HEVC/VP9 HW acceleration improvements.

**Comment by etu** [comment]
As for PHP I think we're over all pretty happy with the current state and don't have any blocking issues.

**Comment by mweinelt** [comment]
I'd like to get Ansible updates in and EOL versions out, they're over in #122670.

**Comment by dotlambda** [comment]
I'd like to get #120440 and #122241 in, as well as #121777 and the fixes necessary for TLS keys to work with the dendrite module using https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Credentials.
cc @mjlbach

**Comment by berbiche** [comment]
It would be great if 21.05 had the some of the latest Erlang/Beam ecosystem changes:
- #122646 introduces Erlang language server (erlang-ls)
- #122723 adds Erlang R24
- #122632 adds rebar3-nix

Maybe even #122633 if possible?

**Comment by vs49688** [comment]
Not a language or framework, but I would like to get this in, as the current version is old:
https://github.com/NixOS/nixpkgs/pull/122140

**Comment by kinnison** [comment]
Since I have been doing some vscode extension updates for myself, #122913 should be uncontroversial and specifically updates a couple of microsoft owned extensions to vscode.  I think the live-share extension is most important in that PR because I fear the risk of incompatibilities.  I confirmed both extensions are behaving for me.  Including doing a live-share between nixos and debian.

**Comment by markuskowa** [comment]
It would be great if we could get the openblas update merged https://github.com/NixOS/nixpkgs/pull/122035.

**Comment by mweinelt** [comment]
I don't want to be forced to set an OnCalendar value in the postgresqlBackup module.

https://github.com/NixOS/nixpkgs/pull/123006

**Comment by risicle** [comment]
I'm finding that building `openblasCompat` from this is causing my darwin machine, which has a setup _very much_ like hydra's, to hang. Hope it doesn't have the same effect on hydra. Not that it's ultimately the fault of _this package_, I think.

**Comment by futile** [comment]
Small version update, but it makes microsoft's `python-language-server` work with Python 3.9, which I think would be appropriate  for 21.05: https://github.com/NixOS/nixpkgs/pull/121522

**Comment by talyz** [comment]
I would like to get these ones in before the freeze:
- #121708 - securely replace secrets in various modules
- #121778 - various `keycloak` module fixes (security related and not) that would be good to have in its first stable release
- #121145 - replace the broken `geoip-updater` module

**Comment by eduardosm** [comment]
Can we get https://github.com/NixOS/nixpkgs/pull/121588 before freeze?
It renames `sip` to `sip_4` and `sip_5` to `sip`. `sip_5` never made it into stable, so if we rename it now we can avoid adding an alias.

**Comment by jonringer** [comment]
For clarification, the original intent of this is for large ecosystems to express anything that's imperative to have in the next release. Most PRs that are targeting master aren't affected by this.

If you just want someone to review your PR, just cc me in the PR please. Or post on the [PRs ready for review thread](https://discourse.nixos.org/t/prs-already-reviewed/2617/)

**Comment by lblasc** [comment]
I would love to see #123074 merged into final staging-next round, LLVM bump was one of the main "features" of rust 1.52 release, https://github.com/rust-lang/rust/blob/master/RELEASES.md#compiler

**Comment by jonringer** [comment]
Any future staging PRs will need to be backported into the release. The backport staging branch won't be created until next week as well.

**Comment by Ericson2314** [comment]
I labeled it, but https://github.com/NixOS/nixpkgs/pull/123415 from @sternenseemann is a fantastic catch of an old bug that somehow slipped underneath the radar for at least one release cycle.

**Comment by tobiasBora** [comment]
Also, I wanted to notify that bluetooth headsets are harder and harder to use (at least on my computer) since 20.09 (just created a bug report [there](https://github.com/NixOS/nixpkgs/issues/123784)): I need to kill pulseaudio & restart applications to be able to switch to bluetooth, audio is muted by default, and since more recently I can't anymore use high fidelity bluetooth output (I'm stuck with the low quality sound+microphone mode). I'm not sure if I'm the only one experiencing such troubles, but I'm thinking that it could be great to fix bluetooth issues before releasing 21.05.

**Comment by domenkozar** [comment]
@tobiasbora pipewire has a lot better support for Bluetooth headsets, so that's your best bet. 

Pulseaudio development was a mess for the last couple of years and pipewire just has better architecture going forward.

**Comment by tobiasBora** [comment]
@domenkozar I can confirm that after a quick test, pipewire has much better support than pulseaudio (and supports much more bluetooth profiles). But I have 3 new issues with pipewire (maybe related):

- Impossible to change volume with the volume keys on the keyboard (on KDE Plasma), I need to use pavucontrol to do so
- The "volume" icon in the systray is not present anymore
- `pactl` command is not present anymore: some people seem to have this command present, and it proves useful to use command lines to change volume, like `pactl set-sink-volume @DEFAULT_SINK@ +10%` or to check that the system is using pipewire and not pulseaudio (with `pactl info`). This command is present in `pulseaudio` packages, but I'm afraid to have conflicts between pulseaudio and pipewire if I install it.

I also tried to add my user to the `audio` group, it does not change anything.

**Comment by asymmetric** [comment]
@tobiasBora as explained by @jonringer above:

> For clarification, the original intent of this is for large ecosystems to express anything that's imperative to have in the next release. Most PRs that are targeting master aren't affected by this.

Please don't post issues you're having with packages here, let's try to keep this focused :)

**Comment by jonringer** [comment]
Seeing as the release is scheduled for tomorrow (will likely be pushed back until `staging-next` jobsets get closer to completing), I'm going to close this issue.

**Comment by yurrriq** [comment]
I'm late to this, but we should really get https://github.com/NixOS/nixpkgs/pull/124300 in 21.05, because https://ferd.ca/you-ve-got-to-upgrade-rebar3.html

**Comment by vcunat** [comment]
Security fixes are never late.  They're the main point of maintaining release branches.

**Comment by neonfuz** [comment]
https://github.com/NixOS/nixpkgs/pull/124776

If this could be included in the 21.05 release then current users using the nixpkgs.config.dwm.patches interface wouldn't end up with a broken system on upgrade
